app/steps/ui/executors/address/index.js

```diff
@@ -22,6 +22,9 @@ class ExecutorAddress extends AddressStep {
         } else if (startsWith(req.path, path)) {
             ctx.index = this.recalcIndex(ctx, 0);
         }
+        if (req.session.addresses && req.session.addresses.executors && req.session.addresses.executors[ctx.index-1]) {
+            ctx.addresses = req.session.addresses.executors[ctx.index-1];
+        }
```

Comment:
Can we have unit tests to test this logic?

Reply:
yep, have now added new test in testExecutorAddress.js called 'should return the ctx with the executors addresses for the correct index'